### ISSUE TITLE
Make figure accept Title instance

### ISSUE THREAD

**Issue by brendancol** [bokeh/bokeh#5318]
Right now the example below throws an error:
```
from bokeh.models import Title
from bokeh.io import show, output_file
from bokeh.plotting import figure

t = Title(text='hello')

output_file('title_example.py')
p = figure(title=t)
show(p)
```

**Comment by birdsarah** [comment]
This is a bug. Line 42 and 50 of bokeh/plotting/figure.py

**Comment by birdsarah** [comment]
@brendancol you may also want tot take a look and see if charts have the same problem.

**Comment by bryevdv** [comment]
@brendancol FYI to get the issue to auto close the PR comment has to day "fixes #1234" or "closes #1234"
